### ISSUE TITLE
Make SystemJS / StealJS support environments that only have `fetch` and no XMLHttpRequestObject

### ISSUE THREAD

**Issue by justinbmeyer** [stealjs/steal#416]
ServiceWorkers do not have XMLHttpRequest objects, so this breaks:

https://github.com/stealjs/steal/blob/master/steal.js#L2647

We need a version of fetchTextFromURL that works against fetch.

**Comment by justinbmeyer** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch

```js
fetchTextFromURL = function(url, fulfill, reject) {
  fetch(url).then(function(response){
   fulfil(response.text()
 }, reject)
}
```

**Comment by airhadoken** [comment]
:+1:

**Comment by matthewp** [comment]
Fixed by #879
